firefox-ios/firefox-ios-tests/Tests/ClientTests/Microsurvey/Mock/MicrosurveyPromptMiddlewareTests.swift

```diff
@@ -25,64 +27,97 @@ final class MicrosurveyPromptMiddlewareTests: XCTestCase {
+        DependencyHelperMock().bootstrapDependencies()
+        setupStore()
     }
 
     override func tearDown() {
         DependencyHelperMock().reset()
+        resetStore()
         super.tearDown()
     }
 
     func testShowPromptAction_withInvalidModel() {
-        DependencyHelperMock().bootstrapDependencies()
-        let mockStore = Store(
-            state: AppState(),
-            reducer: AppState.reducer,
-            middlewares: [MicrosurveyPromptMiddleware().microsurveyProvider]
+        let subject = createSubject(microsurveyManager: MockMicrosurveySurfaceManager(with: nil))
+
+        let action = MicrosurveyPromptMiddlewareAction(
+            windowUUID: .XCTestDefaultUUID,
+            actionType: MicrosurveyPromptActionType.showPrompt
         )
 
-        let action = getAction(for: .showPrompt)
-        mockStore.dispatch(action)
+        subject.microsurveyProvider(AppState(), action)
+
+        XCTAssertEqual(mockStore.dispatchedActions.count, 0)
         XCTAssertEqual(microsurveyManager.handleMessageDisplayedCount, 0)
     }
 
-    func testShowPromptAction_withValidModel() {
-        let mockStore = Store(
-            state: AppState(),
-            reducer: AppState.reducer,
-            middlewares: [MicrosurveyPromptMiddleware().microsurveyProvider]
+    func testShowPromptAction_withValidModel() throws {
+        let subject = createSubject(microsurveyManager: microsurveyManager)
+        let action = MicrosurveyPromptMiddlewareAction(
+            windowUUID: .XCTestDefaultUUID,
+            actionType: MicrosurveyPromptActionType.showPrompt
         )
 
-        let action = getAction(for: .showPrompt)
-        mockStore.dispatch(action)
+        subject.microsurveyProvider(AppState(), action)
+
+        let actionCalled = try XCTUnwrap(mockStore.dispatchedActions.first as? MicrosurveyPromptMiddlewareAction)
+        let actionType = try XCTUnwrap(actionCalled.actionType as? MicrosurveyPromptMiddlewareActionType)
+
+        XCTAssertEqual(actionType, MicrosurveyPromptMiddlewareActionType.initialize)
+        XCTAssertEqual(mockStore.dispatchedActions.count, 1)
         XCTAssertEqual(microsurveyManager.handleMessageDisplayedCount, 1)
     }
 
     func testClosePromptAction() {
-        let mockStore = Store(
-            state: AppState(),
-            reducer: AppState.reducer,
-            middlewares: [MicrosurveyPromptMiddleware().microsurveyProvider]
+        let subject = createSubject(microsurveyManager: microsurveyManager)
+        let action = MicrosurveyPromptMiddlewareAction(
+            windowUUID: .XCTestDefaultUUID,
+            actionType: MicrosurveyPromptActionType.closePrompt
         )
 
-        let action = getAction(for: .closePrompt)
-        mockStore.dispatch(action)
+        subject.microsurveyProvider(AppState(), action)
+        XCTAssertEqual(mockStore.dispatchedActions.count, 0)
```

Comment:
```suggestion
        subject.microsurveyProvider(AppState(), action)

        XCTAssertEqual(mockStore.dispatchedActions.count, 0)
```

Let's just add a bit of spacing between the action we are testing and the test cases. 🙂